### PR TITLE
MP Rest Client SSL support

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/header/ClientHeadersRequestFilter.java [hunk 1]
```diff
@@ -92,8 +90,6 @@ public void filter(ClientRequestContext requestContext) {
                 .forEach(
                         (key, values) -> requestContext.getHeaders().put(key, castToListOfObjects(values))
                 );
-
-        ResteasyProviderFactory.getContextDataMap().put(HttpHeaders.class, new HttpHeadersContextProvider(requestContext));
```
Comment (michalszynkiewicz) [inline resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/header/ClientHeadersRequestFilter.java]: There used to be a test in MP Rest Client TCK that required it but the test is removed now.

**Comment by michalszynkiewicz** [comment]
To be able to get a successful build, I added exclusions for tests that verify features that are not yet implemented.

File: resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/header/ClientHeadersRequestFilter.java (hunk 1 above)
Comment (asoldano) [reply resteasy-client-microprofile/src/main/java/org/jboss/resteasy/microprofile/client/header/ClientHeadersRequestFilter.java]: Great one less thing in the map there.

**Comment by michalszynkiewicz** [comment]
The PR for 3.8: https://github.com/resteasy/Resteasy/pull/2047
This one can be closed, right?

**Comment by michalszynkiewicz** [comment]
Turns out this should be done in 3.8 only, closing.

**Pull request closed without merging**